decompiler/IR2/AtomicOpTypeAnalysis.cpp

```diff
@@ -1433,13 +1433,13 @@ TypeState StackSpillLoadOp::propagate_types_internal(const TypeState& input,
   // stack slot load
   auto& info = env.stack_spills().lookup(m_offset);
   if (info.size != m_size) {
-    env.func->warnings.general_warning(
+    env.func->warnings.general_error(
         "Stack slot load at {} mismatch: defined as size {}, got size {}", m_offset, info.size,
         m_size);
   }
 
   if (info.is_signed != m_is_signed) {
-    env.func->warnings.general_warning("Stack slot offset {} signed mismatch", m_offset);
+    env.func->warnings.general_error("Stack slot offset {} signed mismatch", m_offset);
```

Comment:
this one might be better as a warning - the GOAL compiler actually gets this wrong in a few places